sklearn/impute/_iterative.py

```diff
@@ -707,7 +722,7 @@ def fit_transform(self, X, y=None):
                 Xt, estimator = self._impute_one_feature(
                     Xt,
                     mask_missing_values,
-                    feat_idx,
+                    int(feat_idx),
```

Comment:
We need to understand why `feat_idx` is not an integer here.

Reply:
The `feat_idx` is coming out of `_get_ordered_idx`, where a few different things are done based on the imputation_order. Do you know if the issue happened in general, or for a specific imputation_order?

Reply:
I assume that it was the default one. I will have a closer look to this function.

Reply:
OK the reason to get a numpy array was for the following:

https://github.com/scikit-learn/scikit-learn/blob/75db1bc2dbd716016170241d8daae97624d01252/sklearn/utils/__init__.py#L189-L193

We can remove it.

Reply:
`np.int64(10)` will expose a `shape` attribute :)